### ISSUE TITLE
Panics on start with "stream did not contain valid UTF-8"

### ISSUE THREAD

**Issue by alopatindev** [adder46/hstr-rs#12]
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }', hstr/src/app.rs:40:69
19:     0x55f7f82f299c - std::rt::lang_start::{{closure}}::h7c59ab7a4490bfbc
20:     0x55f7f839adf7 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::he179d32a5d10d957 at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:259
21:     0x55f7f839adf7 - std::panicking::try::do_call::hcb3d5e7be089b2b4 at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:381
22:     0x55f7f839adf7 - std::panicking::try::h7ac93b0cd56fb701 at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:345
23:     0x55f7f839adf7 - std::panic::catch_unwind::h7b40e396c93a4fcd at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:382
24:     0x55f7f839adf7 - std::rt::lang_start_internal::h142b9cc66267fea1 at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/rt.rs:51
25:     0x55f7f82fe142 - main
26:     0x7f52ee1bfd0b - __libc_start_main
27:     0x55f7f82ed0aa - _start
28:                0x0 - <unknown>
```

```
$ uname -or
5.9.14 GNU/Linux
$ rustc --version
rustc 1.48.0 (7eac88abb 2020-11-16)
```

Might relates to #10 (but the OS and stacktrace are different).

Happens with specific history file:
```
$ file .zsh_history
.zsh_history: Non-ISO extended-ASCII text, with very long lines
```

Doesn't reproduce for fresh .zsh_history

**Comment by adder46** [comment]
Is there a chance that you can upload your history file somewhere?

**Comment by adder46** [comment]
Okay, I've diagnosed the issue.

For some reason, zsh stores its history file in zsh's internal metafied format.

Fixes for this and #10 coming soon.
